src/gfx/reverse.cpp

```diff
@@ -171,13 +171,24 @@ void reverse() {
 			        options.nbPalettes);
 		}
 
-		if (options.palSpecType == Options::EXPLICIT && palettes != options.palSpec) {
-			warning("Colors in the palette file do not match those specified with `-c`!");
+		if (options.palSpecType == Options::EXPLICIT) {
+			for (auto& palette : palettes) {
+				for (auto& specPalette : options.palSpec) {
+					for (auto& color : palette) {
+						for (auto& specColor : specPalette) {
```

Comment:
Formatting convention. (Also, lol, that's some deep looping.)
```suggestion
			for (auto &palette : palettes) {
				for (auto &specPalette : options.palSpec) {
					for (auto &color : palette) {
						for (auto &specColor : specPalette) {
```

Reply:
You should be able to call [`std::equal`](https://en.cppreference.com/w/cpp/algorithm/equal) with a custom comparator, then. (That's overload #3.)

Reply:
> It used to be a really simple `palette == specPalette` but that doesn't work now that they're different types.

Are they? `clangd` tells me that both of them are `std::vector<std::array<std::optional<Rgba>, 4>>`.

Reply:
They were different types at the time of me writing that :P